crates/bevy_pbr/src/render/pbr.wgsl

```diff
@@ -145,6 +145,12 @@ fn fragment(
         output_color = pbr_functions::alpha_discard(pbr_bindings::material, output_color);
     }
 
+    /// User clipping, everyting below the plane gets discarded if the length of 
+    /// the normal vector of the plane is non-zero
+    if dot(in.world_position.xyz, view.user_clip.xyz) < view.user_clip.w {
+        discard;
```

Comment:
I tried to do this, but Its not straight forward for me how to do it. I'm still quite new to the bevy render code, but for me it seems like that a shader program is compiled pr Material. This feature is pr. camera, so can I even add a shader def pr camera? I see that the fog function is not in a shader def, which leads me to think that this is because its not straight forward?